device/thunder_ripple_sdk/src/processors/thunder_device_info.rs

```diff
@@ -675,18 +679,28 @@ impl ThunderDeviceInfoRequestProcessor {
             })
             .await;
         info!("{}", response.message);
-        let hdcp_version = response.message["supportedHDCPVersion"].to_string();
-        let is_hdcp_supported: bool = response.message["isHDCPSupported"]
-            .to_string()
-            .parse()
-            .unwrap();
         let mut hdcp_response = HashMap::new();
-        if hdcp_version.contains("1.4") {
-            hdcp_response.insert(HdcpProfile::Hdcp1_4, is_hdcp_supported);
-        }
-        if hdcp_version.contains("2.2") {
-            hdcp_response.insert(HdcpProfile::Hdcp2_2, is_hdcp_supported);
+
+        if let Some(v) = response.message.get("supportedHDCPVersion") {
```

Comment:
For all of these can you just do an early return on None instead of putting the entire method inside a Some block?

Reply:
Followed this suggestion and now ThunderDevice Info has grown in to bit diff. Based on discussion with team will reduce the scope of thunder_device_info to a smaller fix for bootstrap related APIs